Cubical/Categories/Constructions/Coend/Base.agda

```diff
@@ -0,0 +1,104 @@
+{-# OPTIONS --safe #-}
+-- following 1 Lab
+-- https://1lab.dev/Cat.Diagram.Coend.Sets.html
+
+module Cubical.Categories.Constructions.Coend.Base where 
+    open import Cubical.Categories.Bifunctor.Base
+    open import Cubical.Categories.Category
+    open import Cubical.Categories.Functor
+    open import Cubical.Categories.Instances.Sets
+
+    open import Cubical.Foundations.HLevels
+    open import Cubical.Foundations.Prelude
+
+    private
+        variable
+            ℓC ℓC' ℓD ℓD' : Level
+
+    module _ {C : Category ℓC ℓC'}{D : Category ℓD ℓD'}(F : Bifunctor (C ^op) C D)where
+
+        open Category C renaming (ob to obᶜ ; _⋆_ to _⋆ᶜ_ ; id to idᶜ)
+        open Category D renaming (ob to obᵈ ; _⋆_ to _⋆ᵈ_ ; id to idᵈ)
+
+        record Cowedge : Set (ℓ-max ℓC (ℓ-max ℓC' (ℓ-max ℓD ℓD'))) where 
+            field 
+                nadir : obᵈ
+                ψ : (c : obᶜ) → D [ F ⟅ c , c ⟆b , nadir ]
+                {- 
+                    for all morphisms f : c ⇒ c' in category C,
+
+                    F₀(c',c)---F₁(id(c'),f)---> F₀(c',c')
+                     |                          |
+                     F₁(f,id(c))               ψ(c')
+                     ↓                          ↓ 
+                    F₀(c,c)---ψ(c)-----------> nadir
+                -}
+                extranatural : {c c' : obᶜ}(f : C [ c , c' ]) → 
+                    (F ⟪ idᶜ , f ⟫lr ⋆ᵈ  ψ c') ≡ ( F ⟪ f ,  idᶜ ⟫lr ⋆ᵈ ψ c)
+                
+        record Coend : Set ((ℓ-max ℓC (ℓ-max ℓC' (ℓ-max ℓD ℓD'))))  where 
```

Comment:
Can probably define a `Presheaf` structure for `Cowedge` and then this should be isomorphic to a universal element of that presheaf. Don't need to do it now but should include a TODO

Reply:
Added a TODO. 

I should probably try that later. Steven was just teaching me about universal elements today (using products as an example).